src/components/screens/map-screen/map-screen.tsx

```diff
@@ -1049,7 +1058,39 @@ class MapScreen extends React.Component<Props, State> {
         newCenter[1] = newCenterCoordinates[1];
       }
 
-      this.setState({ route, locationFrom, locationTo, loadingRoute: false, mapViewport: { center: newCenter as [number, number], zoom: 13 }, previousZoom: 13, polyline }); 
+      let formattedRoute: string[] = []
+      for (var index in route) {
+        const coordinates = route[index]
+        formattedRoute.push(coordinates[0]+','+coordinates[1])
+      }
+      
+      const airQualityApi = Api.getAirQualityApi(accessToken);
+      const routeAirQuality = await airQualityApi.getAirQualityForCoordinatesArray({ 
+        requestBody: formattedRoute,
+        routingTime: departureTime
+      }
+    );
+      const routePollutionTotals = routeAirQuality.pollutionDataTotals
+      const pollutantsApi = Api.getPollutantsApi(accessToken);
+
+      var routeTotalExposures: RouteTotalExposure[] = [];
+      for (var routePollutionIndex in routePollutionTotals) {
+        let value:number = routePollutionTotals[routePollutionIndex].value || 0;
+        let providedPollId: string = routePollutionTotals[routePollutionIndex].pollutantId || '';
+        if (providedPollId === '') {
+          break;
+        }
+        let pollutantName = await pollutantsApi.findPollutant({
+          pollutantId: providedPollId
+        })
+
+        routeTotalExposures.push({
+          pollutantName: pollutantName.displayName,
+          pollutionValue: value
+        });      
+    }
+      
+      this.setState({ routeTotalExposures, route, locationFrom, locationTo, loadingRoute: false, mapViewport: { center: newCenter as [number, number], zoom: 13 }, previousZoom: 13, polyline }); 
```

Comment:
```suggestion
      this.setState({ route, locationFrom, locationTo, loadingRoute: false, mapViewport: { center: newCenter as [number, number], zoom: 13 }, previousZoom: 13, polyline }); 
```
Also to be moved to a private method